src/Converter.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Money;
+
+final class Converter
+{
+    /**
+     * @var Currencies
+     */
+    private $currencies;
+
+    /**
+     * @param Currencies $currencies
+     */
+    public function __construct(Currencies $currencies)
+    {
+        $this->currencies = $currencies;
+    }
+
+    /**
+     * @param Money        $money
+     * @param CurrencyPair $currencyPair
+     * @param int          $roundingMode
+     *
+     * @return Money
+     */
+    public function convert(Money $money, CurrencyPair $currencyPair, $roundingMode = Money::ROUND_HALF_UP)
+    {
+        if ($money->getCurrency()->getCode() !== $currencyPair->getBaseCurrency()->getCode()) {
+            throw new \InvalidArgumentException('Base Currency of currency pair does not equal money currency');
+        }
+
+        $subunitBaseCurrency = $this->currencies->subunitFor($currencyPair->getBaseCurrency());
+        $subunitCounterCurrency = $this->currencies->subunitFor($currencyPair->getCounterCurrency());
+        $differenceInSubunit = $subunitBaseCurrency - $subunitCounterCurrency;
+        $ratio = $currencyPair->getConversionRatio();
+
+        if ($differenceInSubunit > 0) {
```

Comment:
Maybe I wasn't clear: I would prefer having the solution above as it's essentially the same, but simpler. Is there any specific use case why we need both branches?